source/common/network/address_impl.cc

```diff
@@ -113,7 +81,6 @@ Ipv4Instance::Ipv4Instance(const sockaddr_in* address) : InstanceBase(Type::Ip)
   friendly_name_.append(ip_.friendly_address_);
   friendly_name_.push_back(':');
   friendly_name_.append(port.data(), port.size());
-  validateIpv4Supported(friendly_name_);
```

Comment:
I'm uncomfortable with this part of the change, as I suspect this is being used currently as part of the config verification flow, even if this is (clearly) not well tested right now. Can we potentially back this part of the change out in some way and we can circle back on how to handle this as a discrete PR? cc @ggreenway @htuch @hennna who might have some more context on this.

Reply:
Yep, let's move this out. It would nice to still do graceful failure when the wrong IP address version is used in environments where there is no v4 or v6.

Reply:
Re-adding the code and will submit a separate PR.

Question however is how should we handle these checks? I was thinking we could have some system level Api config and maybe use that here, i.e., figure out only once per envoy instance if v4 and v6 are supported. That would allow us to keep the logic here (without the syscalls) and therefore avoid checking in all the places where addresses are created if a certain version is supported.

Reply:
Can this be part of the OS syscall singleton? I don't think it's a syscall though. Maybe we should have a platform singleton that OS syscalls belongs to. That's a biggish change.

Reply:
I'm actually in the middle of a series of patches that should end up with an "injectable" host stack/socket api that hangs off of `Api` interface (at least that's the high level plan @mattklein123 and I discussed). That virtual class is currently a namespace, i.e., `SocketInterface`, so probably now would be the right time to start looking at converting it. It already has an ipFamilySupported function, so it should be just a matter of shuffling things around (FLW).